config/application.rb

```diff
@@ -40,7 +40,11 @@ class Application < Rails::Application
       allow do
         origins do |source, _env|
           ServiceProvider.pluck(:redirect_uris).flatten.map do |uri|
-            URI.join(uri, '/').to_s[0..-2]
+            begin
+              URI.join(uri, '/').to_s[0..-2]
+            rescue URI::BadURIError
```

Comment:
Do we have any mechanism to show a warning or something? In the past I've seen a convention like `assert_soft()`, which complains but carries on in production, and raises in tests.

Reply:
I'll add a `Rails.logger.warn`

Reply:
Yeah, good point. We would want to know if a bad URI made it through. Maybe the best solution is to add a validation in the ServiceProvider model?